.github/workflows/ci-check-bundle-size.yml

```diff
@@ -0,0 +1,107 @@
+name: Check Binaries bundle size
+
+on:
+  push:
+    branches: [main]
+
+  pull_request:
+    branches: [main]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ (github.event.pull_request && github.event.pull_request.number) || github.ref || github.run_id }}
+  cancel-in-progress: true
+
+# See https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions
+permissions:
+  contents: read
+
+jobs:
+  generate-matrix:
```

Comment:
we don't want to replicate the full build of the binaries here. There's only v2 binary from ./cmd/jaeger that we want to track, and only for one architecture.